crates/re_viewer/src/ui/welcome_screen/welcome_page.rs

```diff
@@ -60,114 +42,183 @@ pub(super) fn welcome_page_ui(
+        - grid_spacing.x)
+        .floor()
+        .at_most(MAX_COLUMN_WIDTH);
 
-        ui.horizontal(|ui| {
-            button_centered_label(ui, "Quick start…");
-            // TODO(ab): activate when C++ is ready!
-            // url_large_text_button(re_ui, ui, "C++", CPP_QUICKSTART);
-            url_large_text_button(re_ui, ui, "Python", PYTHON_QUICKSTART);
-            url_large_text_button(re_ui, ui, "Rust", RUST_QUICKSTART);
-        });
+    // this space is added on the left so that the grid is centered
+    let centering_space = (ui.available_width()
+        - column_count as f32 * column_width
+        - (column_count - 1) as f32 * grid_spacing.x)
+        .max(0.0)
+        / 2.0;
 
-        ui.horizontal(|ui| {
-            if large_text_button(ui, "Open file…").clicked() {
-                command_sender.send_ui(re_ui::UICommand::Open);
-            }
-            button_centered_label(ui, "Or drop a file anywhere!");
-        });
+    ui.horizontal(|ui| {
+        ui.add_space(centering_space);
 
-        ui.horizontal(|ui| {
-            url_large_text_button(re_ui, ui, "Learn about Views", SPACE_VIEWS_HELP);
-        });
+        ui.vertical(|ui| {
+            ui.horizontal_wrapped(|ui| {
+                ui.add(egui::Label::new(
+                    egui::RichText::new("Welcome.")
+                        .strong()
+                        .line_height(Some(32.0))
+                        .text_style(re_ui::ReUi::welcome_screen_h1()),
+                ));
 
-        ui.end_row();
-    });
+                ui.add(egui::Label::new(
+                    egui::RichText::new("Visualize multimodal data.")
+                        .line_height(Some(32.0))
+                        .text_style(re_ui::ReUi::welcome_screen_h1()),
+                ));
+            });
+
+            ui.add_space(32.0);
+
+            let grid = egui::Grid::new("welcome_screen_grid")
+                .spacing(grid_spacing)
+                .min_col_width(column_width)
+                .max_col_width(column_width);
+
+            grid.show(ui, |ui| {
+                let mut show_example = false;
+
+                for panels in panels.chunks(column_count) {
+                    if column_count == 4 {
+                        for panel in panels {
+                            image_banner(re_ui, ui, panel.image, column_width);
+                        }
+                    } else {
+                        for _ in panels {
+                            ui.vertical(|ui| {
+                                ui.add_space(20.0);
+                            });
+                        }
+                    }
+
+                    ui.end_row();
+
+                    for (idx, panel) in panels.iter().enumerate() {
+                        ui.vertical(|ui| {
+                            // don't let the text get too close to the right-hand content, if any
+                            if (idx + 1) % column_count != 0 {
```

Comment:
This would be a lot more readable if you gave the expression a name, i.e.

```
let is_last_column = (idx + 1) % column_count == 0;
if !is_last_column {
```

Reply:
Actually dropping that entire part as it breaks design rules.